### PR TITLE
local tokens can self reference

### PR DESCRIPTION
**Pull request by AlexErrant**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "@types/mocha": "^5.2.6",
     "@types/node": "^20.5.0",
     "ist": "^1.1.1",
-    "mocha": "^10.2.0"
+    "mocha": "^10.2.0",
+    "ts-node": "^10.9.2"
```
Comment (AlexErrant) [inline package.json:35]: I had to add `ts-node` otherwise I couldn't run tests. Not sure if it was just my machine, but whatever.

Comment (AlexErrant) [reply package.json:35]: Probably due to https://github.com/lezer-parser/generator/blob/d827142436db3724af4b4dd9162132ca343f34e0/.mocharc.cjs#L4

**Comment by marijnh** [comment]
I've added the ts-node dependency in df7c098f932f9e23e59d271d4c3ab27ab99e5799

What does "local tokens can self reference" mean? What problem does this PR solve?

**Comment by AlexErrant** [comment]
Sorry I was perhaps too terse in my nonexistent description 😅

In the following:

```
@local tokens {
  Y { "y" }
  Z { "z" }
  Letter { Y | Z }
  @else stringContent
}
```

`Letter` references `Y` and `Z` in the local tokens block. It is "self-referencing". Please see the test for a more complete example.

This PR breaks out of the for loop early if `rule` is found. (In a way you can think of the original behavior as a bug because it doesn't make sense to write `undefined` into `rule` once it has been found.)

**Comment by marijnh** [comment]
That makes sense. Merged!

**Comment by AlexErrant** [comment]
Just wondering, is it worth cutting a release for this? (I can always pnpm patch for my own proj, but I was just wondering if you were holding off for some reason.)

**Comment by marijnh** [comment]
Done!

**Pull request merged**